src/hastings_util.erl

```diff
@@ -42,3 +43,18 @@ callback(complete, Acc) ->
     {ok, lists:reverse(Acc)};
 callback(timeout, _Acc) ->
     {error, timeout}.
+
+
+rename_dir(RootDelDir, Original, DbName) ->
+    GeoIdxDirList = lists:subtract(
```

Comment:
This logic looks wrong to me. The first thing that comes to mind is if there are repeated directory names. I think we usually do this relative offset using a binary:longest_common_prefix or something.